pkg/cloud/services/loadbalancer/loadbalancer.go

```diff
@@ -227,16 +227,26 @@ func (s *Service) getOrUpdateAllowedCIDRS(openStackCluster *infrav1.OpenStackClu
 	if len(openStackCluster.Spec.APIServerLoadBalancer.AllowedCIDRs) > 0 {
 		allowedCIDRs = append(allowedCIDRs, openStackCluster.Spec.APIServerLoadBalancer.AllowedCIDRs...)
 
-		if openStackCluster.Spec.Bastion.Enabled {
-			allowedCIDRs = append(allowedCIDRs, openStackCluster.Status.Bastion.FloatingIP, openStackCluster.Status.Bastion.IP)
-		}
+		// In the first reconciliation loop, only the Ready field is set in openStackCluster.Status
+		// All other fields are empty/nil
+		if openStackCluster.Status.Network != nil {
```

Comment:
just to confirm, from the comment above, so the openStackCluster.Status.Network and openStackCluster.Status.Bastion will both become non-nil so we use openStackCluster.Status.Network as condition in the if statement? otherwise is there any chance that openStackCluster.Status.Network is not nil but openStackCluster.Status.Bastion is nil?